### PR TITLE
chore(client): enable flake8 bugbear for python code

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
- bugbear: https://github.com/PyCQA/flake8-bugbear
- key issues:
  - https://stackoverflow.com/questions/16388059/using-callablex-vs-hasattrx-call
  - https://docs.python-guide.org/writing/gotchas/#late-binding-closures 

## Modules
- [x] Client

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/tests/sdk/test_job_handler.py [hunk 1]
```diff
@@ -1028,7 +1028,7 @@ def report_handler(self): ...
             "mock_user_module:MockReport.report_handler",
         } == {r.name for r in results}
         assert all([r.status == "success" for r in results])
-        assert {1, 1, 10, 1} == {len(r.task_results) for r in results}
+        assert set([1, 1, 10, 1]) == {len(r.task_results) for r in results}
```
Comment (goldenxinxing) [inline client/tests/sdk/test_job_handler.py:1031]: Does it meet expectations？
`set([1,1,10,1]) == {1, 10}`

File: client/starwhale/cli/board/widgets.py [hunk 2]
```diff
@@ -125,12 +125,12 @@ def reload(self) -> None:
         data = self._orderby.sort(self.data)
         for idx, item in enumerate(data):
 
-            def try_render(col: Column) -> t.Any:
-                if col.render:
-                    return col.render(idx, item)
-                return get_field(item, col.key)
+            def try_render(col_: Column, idx_: int, item_: t.Any) -> t.Any:
```
Comment (goldenxinxing) [inline client/starwhale/cli/board/widgets.py:128]: Is there no need to modify it here?

Comment (tianweidut) [reply client/starwhale/cli/board/widgets.py:128]: https://docs.python-guide.org/writing/gotchas/#late-binding-closures

File: client/tests/sdk/test_job_handler.py (hunk 1 above)
Comment (tianweidut) [reply client/tests/sdk/test_job_handler.py]: updated

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**